docs/how-to-build-with-ddn/with-turso.mdx

```diff
@@ -0,0 +1,507 @@
+
+## Overview
+
+This tutorial takes about twenty minutes to complete. You'll learn how to:
+
+- Set up a new Hasura DDN project
+- Connect it to a Turso-hosted database
+- Generate Hasura metadata
+- Create a build
+- Run your first query
+- Create relationships
+- Mutate data
+
+Additionally, we'll familiarize you with the steps and workflows necessary to iterate on your API.
+
+This tutorial assumes you're starting from scratch; you'll connect a hosted Turso instance to Hasura, but you can easily
+follow the steps if you already have data seeded. Hasura will never modify your source schema.
+
+<Prereqs />
+
+## Tutorial
+
+### Step 1. Authenticate your CLI
+
+```sh title="Before you can create a new Hasura DDN project, you need to authenticate your CLI:"
+ddn auth login
+```
+
+This will launch a browser window prompting you to log in or sign up for Hasura DDN. After you log in, the CLI will
+acknowledge your login, giving you access to Hasura Cloud resources.
+
+### Step 2. Scaffold out a new local project
+
+```sh title="Next, create a new local project:"
+ddn supergraph init my-project && cd my-project
+```
+
+Once you move into this directory, you'll see your project scaffolded out for you. You can view the structure by either
+running `ls` in your terminal, or by opening the directory in your preferred editor.
+
+### Step 3. Create and seed a new Turso database
+
+#### Step 3.1. Create a new database
+
+Head to [Turso](https://turso.tech) and create an account if you don't already have one. Then, create a new database
+using their onboarding wizard.
+
+After you finish creating the database, you'll be able to access it from your [dashboard](https://app.turso.tech/).
+
+#### Step 3.2. Create an auth token
+
+From the database, create an auth token and save it.
+
+#### Step 3.3. Seed the database
+
+```sh title="Replacing the db-name, username, and auth token with your values, run the following command to create your first table and seed it with data:"
+curl -L -X POST 'https://<db-name>-<your-username>.turso.io/v2/pipeline' \
+    -H 'Authorization: Bearer <auth-token>' \
+    -H 'Content-Type: application/json' \
+    -d '{
+      "requests": [
+        {
+          "type": "execute",
+          "stmt": {
+            "sql": "CREATE TABLE IF NOT EXISTS users (id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT NOT NULL, age INTEGER NOT NULL)"
+          }
+        },
+        {
+          "type": "execute",
+          "stmt": {
+            "sql": "INSERT INTO users (name, age) VALUES (\"Alice\", 25)"
+          }
+        },
+        {
+          "type": "execute",
+          "stmt": {
+            "sql": "INSERT INTO users (name, age) VALUES (\"Bob\", 30)"
+          }
+        },
+        {
+          "type": "execute",
+          "stmt": {
+            "sql": "INSERT INTO users (name, age) VALUES (\"Charlie\", 35)"
+          }
+        },
+        {
+          "type": "close"
+        }
+      ]
+    }'
```

Comment:
If we pipe this to jq we can get a pretty printed response:

```suggestion
    }' | jq '.'
```

Although up to you.